### PR TITLE
2350023 dropdown selector

### PR DESCRIPTION
**Pull request by kmoll**

### CODE REVIEW

File: tests/modules/entity_browser_test/config/install/entity_browser.browser.test_dropdown.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    label: 'View widget nr. 2'
+    weight: 1
+    uuid: df9b03c8-4ad4-44c9-925a-57ab0cc7e6d6
+    settings: {  }
+  a1665840-4b0b-11e4-916c-0800200c9a66:
+    id: upload
+    label: 'Upload'
+    weight: 0
+    uuid: a1665840-4b0b-11e4-916c-0800200c9a66
+    settings: {  }
```
Comment (slashrsm) [inline tests/modules/entity_browser_test/config/install/entity_browser.browser.test_dropdown.yml:28]: Missing newline.

File: src/Tests/EntityBrowserTest.php [hunk 2]
```diff
@@ -238,6 +238,30 @@ protected function testDynamicRoutes() {
     $this->assertEqual($registered_route->getDefault('_content'), 'Drupal\entity_browser\Controllers\StandalonePage::page', 'Controller matches.');
     $this->assertEqual($registered_route->getDefault('_title_callback'), 'Drupal\entity_browser\Controllers\StandalonePage::title', 'Title callback matches.');
     $this->assertEqual($registered_route->getRequirement('_permission'), 'access ' . String::checkPlain($entity->id()) . ' entity browser pages', 'Permission matches.');
+
+    /** @var $entity \Drupal\entity_browser\EntityBrowserInterface */
```
Comment (slashrsm) [inline src/Tests/EntityBrowserTest.php:242]: We probably don't need to check routes for all test browsers that we have. If we test for one we can be pretty sure it works correctly for others too.

File: src/Plugin/EntityBrowser/WidgetSelector/DropDown.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    );
+
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCurrentWidget(WidgetsCollection $widgets) {
+    return $this->getFirstWidget($widgets->sort());
```
Comment (slashrsm) [inline src/Plugin/EntityBrowser/WidgetSelector/DropDown.php:44]: I do not understand entirely how this works. I imagined we need to submit form when user changes value in dropdown and remember selection in this plugin. Then we return that selection from this function. Makes sense?

Comment (kmoll) [inline src/Plugin/EntityBrowser/WidgetSelector/DropDown.php:44]: Yes, I think the getCurrentWidget function is incorrect, it shouldn't return the first widget, there should be a currentWidget property and this function just should return that.  The submit function should just set the current widget.  Thoughts?

Comment (slashrsm) [inline src/Plugin/EntityBrowser/WidgetSelector/DropDown.php:44]: Yes, that sounds right.

**Pull request closed without merging**